source/base/Promise.ooc

```diff
@@ -103,6 +104,13 @@ _ThreadResultPromise: class <T> extends ResultPromise<T> {
 		status := this wait()
```

Comment:
Would `(status := this wait()) ? this _result[T] : defaultValue` work? Just curious.